extensions/shared/register-jetpack-block.js

```diff
@@ -44,9 +57,23 @@ export default function registerJetpackBlock( name, settings, childBlocks = [] )
 		return false;
 	}
 
+	let blockTitle = settings.title;
+	const blockTags = [];
+
+	if ( requiredPlan ) {
+		blockTags.push( availableBlockTags.plan );
+	}
+	if ( betaExtensions.includes( name ) ) {
+		blockTags.push( availableBlockTags.beta );
+	}
+
+	if ( blockTags.length ) {
+		blockTitle = blockTitle + ` (${ blockTags.join( ', ' ) })`;
```

Comment:
How about using template literals all the way to stay consistent?

```suggestion
		blockTitle = `${ blockTitle } (${ blockTags.join( ', ' ) })`;
```